packages/colorpickers/src/elements/ColorPicker/index.tsx

```diff
@@ -72,73 +70,99 @@ export interface IColorPickerProps
-    const [alphaInput, setAlphaInput] = useState<string | number>(state.alpha);
+      previousComputedColorRef.current = controlledColor;
+    }, [controlledColor]);
 
-    const handleColorWellChange = useCallback(
-      (hsv: IHSVColor) => {
-        const hsl = hsvToHsl(hsv.h, hsv.s * 100, hsv.v * 100);
-        const rgb = hslToRgb(hsl.h, hsl.s, hsl.l);
-        const hex = rgbToString(rgb.r, rgb.g, rgb.b);
+    const handleColorWellChange = useCallback((hsv: IHSVColor) => {
+      dispatch({
+        type: 'SATURATION_CHANGE',
+        payload: hsv
+      });
+    }, []);
 
-        if (isControlled) {
-          onChange &&
-            onChange({
-              hue: hsv.h,
-              saturation: hsl.s,
-              lightness: hsl.l,
-              hex,
-              red: rgb.r,
-              green: rgb.g,
-              blue: rgb.b,
-              alpha: state.alpha
-            });
-        } else {
-          dispatch({
-            type: 'SATURATION_CHANGE',
-            payload: hsv
-          });
-        }
-      },
-      [isControlled, onChange, state.alpha]
-    );
+    const handleHueChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+      dispatch({ type: 'HUE_CHANGE', payload: e.target.value });
+    }, []);
 
-    useEffect(() => {
-      setHexInput(state.hex);
-      setRedInput(state.red);
-      setGreenInput(state.green);
-      setBlueInput(state.blue);
-      setAlphaInput(state.alpha);
-    }, [state.hex, state.red, state.green, state.blue, state.alpha]);
+    const handleAlphaSliderChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+      dispatch({ type: 'ALPHA_SLIDER_CHANGE', payload: e.target.value });
+    }, []);
+
+    const handleHexChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+      dispatch({ type: 'HEX_CHANGE', payload: e.target.value });
+    }, []);
+
+    const handleRedChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+      dispatch({ type: 'RED_CHANGE', payload: e.target.value });
+    }, []);
+
+    const handleGreenChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+      dispatch({ type: 'GREEN_CHANGE', payload: e.target.value });
+    }, []);
+
+    const handleBlueChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+      dispatch({ type: 'BLUE_CHANGE', payload: e.target.value });
+    }, []);
+
+    const handleAlphaChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+      dispatch({ type: 'ALPHA_CHANGE', payload: e.target.value });
+    }, []);
+
+    const handleBlur = useCallback(() => {
+      dispatch({ type: 'RESET_COLOR', payload: controlledColor });
+    }, [controlledColor]);
 
     return (
       <StyledColorPicker ref={ref} {...props}>
         <ColorWell
-          hue={state.hue}
-          saturation={state.saturation}
-          lightness={state.lightness}
+          hue={controlledColor.hue}
```

Comment:
All of these value pass-throughs could probably be simplified with a unified `colo={IColor}` prop, but that is unrelated to state management.